diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-a64-wiipiido-wifi.dts b/arch/arm64/boot/dts/allwinner/overlay/sun50i-a64-wiipiido-wifi.dts
new file mode 100644
index 000000000..9785f8836
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-a64-wiipiido-wifi.dts
@@ -0,0 +1,28 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "allwinner,sun50i-a64";
+
+
+	fragment@1 {
+		target = <&mmc1>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			/* Added for WIIPIIDO*/
+			brcmf: wifi@1 {
+                		reg = <1>;
+                		compatible = "brcm,bcm4329-fmac";
+                		interrupt-parent = <&r_pio>;
+                		interrupts = <0 3 8>; /* PL3 */
+                		interrupt-names = "host-wake";
+        		};
+				
+
+		};
+	};
+
+};
+
