diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pine64.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-pine64.dts
index 204c5a7bb..0990b1263 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pine64.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pine64.dts
@@ -350,7 +350,6 @@
 	status = "okay";
 
 	bluetooth {
-		/*compatible = "realtek,rtl8723bs-bt";*/
 		compatible = "brcm,bcm43438-bt";/* for WIIPIIDO  */
 		max-speed = <1500000>; /* for WIIPIIDO  */
 		clocks = <&rtc 1>; /* for WIIPIIDO  */
@@ -360,8 +359,8 @@
 		shutdown-gpios = <&r_pio 0 4 GPIO_ACTIVE_HIGH>; /* PL4 for WIIPIIDO */
 		device-wake-gpios = <&r_pio 0 5 GPIO_ACTIVE_HIGH>; /* PL5 */
 		host-wake-gpios = <&r_pio 0 6 GPIO_ACTIVE_HIGH>; /* PL6 */
-		/*firmware-postfix = "pine64";*/ /* for WIIPIIDO  */
 	};
+
 };
 
 /* On Pi-2 connector */
